encodings/dict/src/array.rs

```diff
@@ -67,11 +68,52 @@ impl ArrayTrait for DictArray {}
 
 impl IntoCanonical for DictArray {
     fn into_canonical(self) -> VortexResult<Canonical> {
-        let canonical_values: Array = self.values().into_canonical()?.into();
-        take(canonical_values, self.codes())?.into_canonical()
+        match self.dtype() {
+            DType::Utf8(_) | DType::Binary(_) => canonicalize_string(self),
+            _ => canonicalize_primitive(self),
+        }
     }
 }
 
+/// Canonicalize a set of codes and values.
+fn canonicalize_string(array: DictArray) -> VortexResult<Canonical> {
+    let values = array.values().into_varbinview()?;
```

Comment:
I guess `into_varbinview()` runs a canonicalize internally? The name of this function doesn't scream quite how expensive it is I think?

Reply:
We made https://github.com/spiraldb/vortex/issues/507. Per https://rust-lang.github.io/api-guidelines/naming.html#ad-hoc-conversions-follow-as_-to_-into_-conventions-c-conv this should likely be `to_varbinview()` but also since this is owned -> owned conversion the docs suggest `into_`, however, `to_` in Rust is meant to indicate expensive conversions